### PR TITLE
STT service improvements

### PR DESCRIPTION
**Pull request by GiviMAD**
googlestt/voskstt: change default maxSilenceSeconds to 3
watsonstt: add singleUtterance mode, rename inativityTimeout to maxSilenceSeconds and minor improvements

### CODE REVIEW

**Comment by GiviMAD** [comment]
Proposed solution to https://github.com/openhab/openhab-addons/issues/12451.

@lolodomo the build failure is not related to the changes.

**Comment by lolodomo** [comment]
Finally, when not in single utterance mode  is 3 seconds enough? WDYT?

File: bundles/org.openhab.voice.watsonstt/src/main/resources/OH-INF/i18n/watsonstt_it.properties [hunk 1]
```diff
@@ -6,8 +6,8 @@ voice.config.watsonstt.group.authentication.label = Autenticazione
 voice.config.watsonstt.group.authentication.description = Informazioni per la connessione all'istanza Watson Speech-to-Text.
 voice.config.watsonstt.group.stt.label = Configurazione STT
 voice.config.watsonstt.group.stt.description = Parametri per l'API Watson Speech-to-Text.
-voice.config.watsonstt.inactivityTimeout.label = Timeout Inattività
-voice.config.watsonstt.inactivityTimeout.description = Il tempo in secondi dopo il quale, se viene rilevato solo silenzio (nessun discorso) nell'audio, la connessione viene chiusa.
+voice.config.watsonstt.maxSilenceSeconds.label = Timeout Inattività
```
Comment (lolodomo) [inline bundles/org.openhab.voice.watsonstt/src/main/resources/OH-INF/i18n/watsonstt_it.properties:9]: To be removed. 
To be done later in Crowdin.

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by GiviMAD** [comment]
> Finally, when not in single utterance mode is 3 seconds enough? WDYT?

Yes I think it's enough, it's the value that I was using. 
Also as the singleUtteranceMode is enabled by default I think the user will tune it as desired when disabling the singleUtteranceMode.

File: bundles/org.openhab.voice.watsonstt/src/main/resources/OH-INF/i18n/watsonstt_it.properties (hunk 1 above)
Comment (GiviMAD) [reply bundles/org.openhab.voice.watsonstt/src/main/resources/OH-INF/i18n/watsonstt_it.properties:9]: So, should I avoid modify property files other than the English one?

File: bundles/org.openhab.voice.watsonstt/src/main/java/org/openhab/voice/watsonstt/internal/WatsonSTTService.java [hunk 2]
```diff
@@ -130,16 +132,16 @@ public STTServiceHandle recognize(STTListener sttListener, AudioStream audioStre
...
                 .build();
         final AtomicReference<@Nullable WebSocket> socketRef = new AtomicReference<>();
         final AtomicBoolean aborted = new AtomicBoolean(false);
         executor.submit(() -> {
             int retries = 2;
             while (retries > 0) {
                 try {
-                    socketRef.set(speechToText.recognizeUsingWebSocket(wsOptions,
-                            new TranscriptionListener(sttListener, config, aborted)));
+                    int internalRetries = 0;
```
[the remark this answers is not in the document]
Comment (GiviMAD) [reply bundles/org.openhab.voice.watsonstt/src/main/java/org/openhab/voice/watsonstt/internal/WatsonSTTService.java:143]: > This looks to me as a possible workaround to a wrong implementation, if required. WDYT?

Yes, it's a workaround for a problem that I'm facing with this SDK from the beginning. 
After the first execution, a SSLPeerUnverifiedException is thrown. The cause seems to be that the underling library used for that (bouncycastle I think) seems to be unloaded after the first execution.
I'm been unable to identify what is causing the issue.

I saw there is some available updates in the dependencies, I will check if that changes something.

File: bundles/org.openhab.voice.watsonstt/src/main/resources/OH-INF/i18n/watsonstt_it.properties (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.voice.watsonstt/src/main/resources/OH-INF/i18n/watsonstt_it.properties:9]: Yes

File: bundles/org.openhab.voice.watsonstt/src/main/java/org/openhab/voice/watsonstt/internal/WatsonSTTService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (GiviMAD) [reply bundles/org.openhab.voice.watsonstt/src/main/java/org/openhab/voice/watsonstt/internal/WatsonSTTService.java:143]: After a little nightmare I discovered the problem, a bug on the java version I was using with TLSv3, described (here)[https://stackoverflow.com/questions/55161904/javax-net-ssl-sslhandshakeexception-from-ibms-watson-assistant-api].
I updated to the last java 11 version available and the problem is solved, so I will remove both retry mechanisms.

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**